drivers/source/BufferedSerial.cpp

```diff
@@ -0,0 +1,382 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2006-2019 ARM Limited
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "drivers/BufferedSerial.h"
+
+#if (DEVICE_SERIAL && DEVICE_INTERRUPTIN)
+
+#include "platform/mbed_poll.h"
+#include "platform/mbed_thread.h"
+
+namespace mbed {
+
+BufferedSerial::BufferedSerial(PinName tx, PinName rx, int baud):
+    SerialBase(tx, rx, baud)
+{
+    enable_rx_irq();
+}
+
+BufferedSerial::BufferedSerial(const serial_pinmap_t &static_pinmap, int baud):
+    SerialBase(static_pinmap, baud)
+{
+    enable_rx_irq();
+}
+
+BufferedSerial::~BufferedSerial()
+{
+    delete _dcd_irq;
+}
+
+void BufferedSerial::dcd_irq()
+{
+    wake();
+}
+
+void BufferedSerial::set_baud(int baud)
+{
+    SerialBase::baud(baud);
+}
+
+void BufferedSerial::set_data_carrier_detect(PinName dcd_pin, bool active_high)
+{
+    delete _dcd_irq;
+    _dcd_irq = NULL;
+
+    if (dcd_pin != NC) {
+        _dcd_irq = new InterruptIn(dcd_pin);
+        if (active_high) {
+            _dcd_irq->fall(callback(this, &BufferedSerial::dcd_irq));
+        } else {
+            _dcd_irq->rise(callback(this, &BufferedSerial::dcd_irq));
+        }
+    }
+}
+
+void BufferedSerial::set_format(int bits, Parity parity, int stop_bits)
+{
+    api_lock();
+    SerialBase::format(bits, parity, stop_bits);
+    api_unlock();
+}
+
+#if DEVICE_SERIAL_FC
+void BufferedSerial::set_flow_control(Flow type, PinName flow1, PinName flow2)
+{
+    api_lock();
+    SerialBase::set_flow_control(type, flow1, flow2);
+    api_unlock();
+}
+#endif
+
+int BufferedSerial::close()
+{
+    /* Does not let us pass a file descriptor. So how to close ?
+     * Also, does it make sense to close a device type file descriptor*/
+    return 0;
+}
+
+int BufferedSerial::isatty()
+{
+    return 1;
+
+}
+
+off_t BufferedSerial::seek(off_t offset, int whence)
+{
+    /* lseek can be done theoratically, but is it sane to mark positions on
+     * a dynamically growing/shrinking buffer system (from an interrupt
+     * context)
+     */
+    return -ESPIPE;
+}
+
+int BufferedSerial::sync()
+{
+    api_lock();
+
+    while (!_txbuf.empty()) {
+        api_unlock();
+        // Doing better than wait would require TxIRQ to also do wake() when
```

Comment:
As mentioned above, this is exactly as it is in `UARTSerial`.

Reply:
Does that mean we shall not fix it ?

Reply:
Changes are not within the scope of the work scheduled.

Reply:
This is a small cleanup and so it should be done.

Reply:
@0xc0170 Where can I find documentation regarding comments?
I thought it would be [here](https://os.mbed.com/docs/mbed-os/v5.15/contributing/style.html#coding-style).